versions/3.0.4.md

```diff
@@ -1068,11 +1068,13 @@ Field Name | Type | Description
 ---|:---:|---
 <a name="parameterStyle"></a>style | `string` | Describes how the parameter value will be serialized depending on the type of the parameter value. Default values (based on value of `in`): for `query` - `form`; for `path` - `simple`; for `header` - `simple`; for `cookie` - `form`.
 <a name="parameterExplode"></a>explode | `boolean` | When this is true, parameter values of type `array` or `object` generate separate parameters for each value of the array or key-value pair of the map. For other types of parameters this property has no effect. When [`style`](#parameterStyle) is `form`, the default value is `true`. For all other styles, the default value is `false`.
-<a name="parameterAllowReserved"></a>allowReserved | `boolean` | Determines whether the parameter value SHOULD allow reserved characters, as defined by [RFC3986](https://tools.ietf.org/html/rfc3986#section-2.2) `:/?#[]@!$&'()*+,;=` to be included without percent-encoding. This property only applies to parameters with an `in` value of `query`. The default value is `false`.
+<a name="parameterAllowReserved"></a>allowReserved | `boolean` | When this is true, parameter values are serialized  using reserved expansion, as defined by [RFC6570](https://datatracker.ietf.org/doc/html/rfc6570#autoid-20), which allows [RFC3986's reserved character set](https://datatracker.ietf.org/doc/html/rfc3986#autoid-13), as well as percent-encoded triples, to pass through unchanged, while still percent-encoding all other disallowed characters (including `%` outside of percent-encoded triples). Applications are still responsible for percent-encoding reserved characters that are not allowed in the query string, or have a meaning other than their literal value (e.g. `+` as an escape for the space character in `application/x-www-form-urlencoded` query strings). This property only applies to parameters with an `in` value of `query`. The default value is `false`.
```

Comment:
The upshot is that if you use `allowReserved: true`, you still need to manually percent-encode the following reserved characters:

* `#` because it is the delimiter between the query and fragment
* `[` and `]` because while they are reserved, they are forbidden in the query by RFC3986 (yes, `deepObject` examples are wrong, see #3859 for more on this)
* `+` because `form-urlencoded` treats it as a space
* `&` and `=` because `form-urlencoded` uses them as delimiters

You can leave any of the others unencoded, and you MUST leave a `+` that represents a space unencoded.  Which maybe I did not show in the example and should have... hm....

Reply:
@darrelmiller I'm open to suggestion on how to better state this.  At one point I had a short paragraph after thist table going into a bit more detail, and then moved it all to the appendix.  Perhaps a short paragraph is warranted, though.